src/network/netdev/fou-tunnel.c

```diff
@@ -112,6 +113,36 @@ static int netdev_fou_tunnel_create(NetDev *netdev) {
         return 0;
 }
 
+int config_parse_ip_protocol(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint8_t *protocol = data;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = parse_ip_protocol(rvalue);
+        if (r < 0 || r > 255)
+                return config_parse_uint8(unit, filename, line, section, section_line,
```

Comment:
hmm, is it really worth it chaining up to config_parse_uint8() here? why not just calle safe_atou8() here on our own and then printing a specific error message if we can't parse things? i mean, if we have a specific parser for this we might as well output a nice error message instead of dropping back to the generic implementatiion again

Reply:
OK. I was just lazy to write a specific error message here. Updated.